lib/components/molecules/Carousel/Carousel.js

```diff
@@ -0,0 +1,398 @@
+
+const CarouselContext = React.createContext([]);
+
+/**
+ * Add unique Index to all slides.
+ *
+ * @param {Array} slides
+ *   all slides
+ *
+ * @return All slides with added slideIndex key as prop.
+ */
+const slideWithIndexKey = slides => {
+  let i = 0;
+  const children = React.Children.map(slides, slide => {
+    i += 1;
+    return React.cloneElement(slide, {
+      id: `slide_${i}`,
+    });
+  });
+
+  return children;
+};
+
+/**
+ * Rendering list of dots
+ */
+function getSliderDots(btnType) {
+  const carouselContent = useContext(CarouselContext);
+  const totalSlidesCount = carouselContent.slides.length;
+  const currentIndex = carouselContent.slideIndex;
+  const setItem = carouselContent.setSlideIndex;
+  const countElements = [];
+
+  for (let id = 1; id <= totalSlidesCount; id += 1) {
+    countElements.push(
+      <li key={id} className={classnames('atomic-item', { active: id === currentIndex })}>
+        <Button onClick={() => setItem(id)} className={classnames('icon', btnType)}>
+          {id}
+        </Button>
+      </li>
+    );
+  }
+
+  return countElements;
+}
+
+/**
+ * Rendering wrapper of dots
+ *
+ * @return HTML
+ *   Wrapper with UL | OL
+ */
+const getList = btnType => {
+  return <ListVanilla>{getSliderDots(btnType)}</ListVanilla>;
+};
+
+/**
+ * Increase the slide count.
+ *
+ * @param {Number} setIndex
+ *   Increase the count of slide.
+ * @param {Number} currentIndex
+ *   Show current active slide number.
+ */
+const handleNext = (setIndex, currentIndex) => {
+  setIndex(currentIndex + 1);
+};
+
+/**
+ * Decrease the slide count.
+ *
+ * @param {Number} setIndex
+ *   Decrease the count of slide.
+ * @param {Number} currentIndex
+ *   Show current active slide number.
+ */
+const handlePrevious = (setIndex, currentIndex) => {
+  setIndex(currentIndex - 1);
+};
+
+/**
+ * Increase slide count on key press over next navigation button.
+ *
+ * @param {Object} e
+ *   event parameter
+ * @param {function} setIndex
+ *   increase the slide index.
+ * @param {Number} currentIndex
+ *   Show current active slide number.
+ */
+const handleKeyNext = (e, setIndex, currentIndex) => {
+  if (e.key === 'ArrowRight') {
+    handleNext(setIndex, currentIndex);
+  }
+
+  return null;
+};
+
+/**
+ * Decrease slide count on key press over previous navigation button.
+ *
+ * @param {Object} e
+ *   event parameter
+ * @param {function} setIndex
+ *   Increase the slide index.
+ * @param {Number} currentIndex
+ *   Show current active slide number.
+ */
+const handleKeyPrev = (e, setIndex, currentIndex) => {
+  if (e.key === 'ArrowLeft') {
+    handlePrevious(setIndex, currentIndex);
+  }
+
+  return null;
+};
+
+/**
+ * Provides HTML for next navigation button.
+ *
+ * @param {Object} setIndex
+ *   Change the slide number.
+ * @param {Number} currentIndex
+ *   Current slide number.
+ * @param {Number} slideCount
+ *   Total number of slides.
+ */
+const nextKey = (setIndex, currentIndex, slideCount, children) => {
```

Comment:
Let's merge this nextKey and prevKey and handle their functional logic through Type. 

use better names get rid of Key its confusing.